flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/GraphCsvReader.java

```diff
@@ -0,0 +1,388 @@
+	{
+
+			this(new Path(Preconditions.checkNotNull(path2, "The file path may not be null.")),mapper, context);
+
+
+	}
+
+	public CsvReader getEdgeReader()
+	{
+		return this.EdgeReader;
+	}
+
+	public CsvReader getVertexReader()
+	{
+		return this.VertexReader;
+	}
+	//--------------------------------------------------------------------------------------------------------------------
+
+	/**
+	 * Specifies the types for the Graph fields and returns a Graph with those field types
+	 *
+	 * This method is overloaded for the case in which Vertices don't have a value
+	 *
+	 * @param type0 The type of CSV field 0 and 1 for the CSV reader used for reading Edge data, the type of CSV field 0 for the CSV reader used for reading Vertex data  and the type of Vetex ID in the returned Graph.
+	 * @param  type1 The type of CSV field 1 for the CSV reader used for reading Vertex data and the type of Vertex Value in the returned Graph.
+	 * @param  type2 The type of CSV field 2 for the CSV reader used for reading Edge data and the type of Edge Value in the returned Graph.
+	 * @return The {@link org.apache.flink.graph.Graph} with Edges and Vertices extracted from the parsed CSV data.
+	 */
+	public <K,VV,EV> Graph<K,VV,EV> types(Class<K> type0, Class<VV> type1, Class<EV> type2)
+	{
+		DataSet<Tuple3<K,K,EV>> edges = this.EdgeReader.types(type0,type0,type2);
+		if(path1!=null)
+		{
+			DataSet<Tuple2<K,VV>> vertices = this.VertexReader.types(type0,type1);
+			return Graph.fromTupleDataSet(vertices,edges,executionContext);
+		}
+		else
+		{
+			return Graph.fromTupleDataSet(edges,mapper,executionContext);
+		}
+
+
+	}
+	/**
+	 * Specifies the types for the Graph fields and returns a Graph with those field types
+	 *
+	 * This method is overloaded for the case in which Vertices don't have a value
+	 *
+	 * @param type0 The type of CSV field 0 and 1 for the CSV reader used for reading Edge data and the type of Vetex ID in the returned Graph.
+	 * @param  type1 The type of CSV field 2 for the CSV reader used for reading Edge data and the type of Edge Value in the returned Graph.
+	 * @return The {@link org.apache.flink.graph.Graph} with Edge extracted from the parsed CSV data and Vertices mapped from Edges with null Value.
+	 */
+	public <K, EV> Graph<K,NullValue,EV> types(Class<K> type0, Class<EV> type1)
+	{
+		DataSet<Tuple3<K,K,EV>> edges = this.EdgeReader.types(type0,type0,type1);
+		return Graph.fromTupleDataSet(edges,executionContext);
+	}
+
+	/**
+	 *Configures the Delimiter that separates rows for the CSV readers used to read the edges and vertices
+	 *	({@code '\n'}) is used by default.
+	 *
+	 *@param delimiter The delimiter that separates the rows.
+	 * @return The GraphCsv reader instance itself, to allow for fluent function chaining.
+	 */
+	public GraphCsvReader lineDelimiter(String delimiter)
+	{
+		this.EdgeReader.lineDelimiter(delimiter);
+		this.VertexReader.lineDelimiter(delimiter);
```

Comment:
when you create a Graph with edges file only, your VertexReader will be null and this will give you a NullPointerException :)